### DISCUSSION TITLE
LoRa & FSK

### DISCUSSION THREAD

**Discussion by lvplummer** [jgromes/RadioLib#850] [General]
Is it possible to instatiate two logical radios - one LoRa and one FSK - connected to the same physical radio?  I can do this now by issuing a new radio.begin for each mode.  However, that takes a lot of time to switch and my application is battery powered which means battery use.

**Comment by jgromes** [comment]
That's not going to work, because the two modes require different configuration, once you enable one, you disable the other. This is the case for both SX127x and SX126x.

**Comment by lvplummer** [comment]
Thanks.

**Comment by lvplummer** [comment]
I have primarily migrated from SX1276 to SX1262 radios for several reasons
unrelated to this discussion.  As I mentioned in my post, I can switch
between LoRa and FSK modes by initializing a new radio with a begin or
beginFSK.  However, battery drain caused by the time required for either
begin is a problem.

I just did a comparison of the time required for a begin for both SX1276
and SX1262 radios using an Arduino Pro Mini CPU.  The SX1262 takes over 250
milliseconds while the SX1276 takes only 18 which I can live with.  Since I
prefer to use the SX1262 radio, I wonder if the much longer SX1262 begin
time is necessary?

**Reply by jgromes** [reply]
That's odd, do you perhaps have debug enabled by mistake? For me, SX1268::begin takes about 34 ms on Arduino Uno.

**Comment by lvplummer** [comment]
I believe that I have debug commented out.

I am probably completely wrong but it seems that begin calls standby which
includes a 100 ms delay.

**Reply by jgromes** [reply]
That delay should only get triggered if the first call to standby fails. You can try to comment it out, but as far as I can tell there should be nothing taking 200+ ms.

**Comment by lvplummer** [comment]
When I change the delay in standby from 100 to 10 ms, the begin goes from
250 ms to 78 ms.  I can live with that.

**Comment by lvplummer** [comment]
With the delay in standby set to 100 ms, the total begin is 256 ms.  When I
change the delay to 10 ms, the begin goes to 76.  I read the arithmetic to
show that the delay is included twice - 256 - (90 x 2) = 76.  90 x 2 is the
total delay reduction when going from 100 to 10.

**Reply by jgromes** [reply]
That's interesting, thanks for investigating. My guess is that for some reason, teh SX126x does not initialize correctly right away, hence why the delay was getting triggered. Still strange it doesn't happen on my Uno which should be pretty close to Pro Mini performance-wise.

I updated the library to only use 10 ms delay, there is no reason to wait 100 ms.

**Comment by lvplummer** [comment]
Thanks.
